src/Endpoints/Delegates/HandlesSettings.php

```diff
@@ -98,20 +99,14 @@ public function resetStopWords(): array
 
     // Settings - Synonyms
 
-    public function getSynonyms(): array
+    public function getSynonyms(): Synonyms
     {
-        return $this->http->get(self::PATH.'/'.$this->uid.'/settings/synonyms');
+        return new Synonyms($this->http->get(self::PATH.'/'.$this->uid.'/settings/synonyms'));
```

Comment:
I would prefer to get an array instead of an Synonyms object here. This change would break the code of users that expect an array.
We could return the array form like this:

```suggestion
        return (new Synonyms($this->http->get(self::PATH.'/'.$this->uid.'/settings/synonyms')))->getIterator()->getArrayCopy();
```

Reply:
Hi @mmachatschek regarding this, Synonyms being an ArrayAccess class will prevent the code to break, cause the client can loop and use it as an array right?